src/Internal/Service/Level3Service.php

```diff
@@ -0,0 +1,192 @@
+		};
+		$items_to_send = array_map( $process_item, $order_items );
+
+		if ( count( $items_to_send ) > 200 ) {
+			// If more than 200 items are present, bundle the last ones in a single item.
+			$items_to_send = array_merge(
+				array_slice( $items_to_send, 0, 199 ),
+				[ $this->bundle_level3_data_from_items( array_slice( $items_to_send, 200 ) ) ]
+			);
+		}
+
+		$level3_data = [
+			'merchant_reference' => (string) $order->get_id(), // An alphanumeric string of up to  characters in length. This unique value is assigned by the merchant to identify the order. Also known as an “Order ID”.
+			'customer_reference' => (string) $order->get_id(),
+			'shipping_amount'    => $this->prepare_amount( (float) $order->get_shipping_total() + (float) $order->get_shipping_tax(), $currency ), // The shipping cost, in cents, as a non-negative integer.
+			'line_items'         => $items_to_send,
+		];
+
+		// The customer’s U.S. shipping ZIP code.
+		$shipping_address_zip = $order->get_shipping_postcode();
+		if ( WC_Payments_Utils::is_valid_us_zip_code( $shipping_address_zip ) ) {
+			$level3_data['shipping_address_zip'] = $shipping_address_zip;
+		}
+
+		// The merchant’s U.S. shipping ZIP code.
+		$store_postcode = get_option( 'woocommerce_store_postcode' );
+		if ( WC_Payments_Utils::is_valid_us_zip_code( $store_postcode ) ) {
+			$level3_data['shipping_from_zip'] = $store_postcode;
+		}
+
+		return $level3_data;
+	}
+
+	/**
+	 * Processes a single order item.
+	 * Based on the queried items, this class should only receive
+	 * `WC_Order_Item_Product` or `WC_Order_Item_Fee` line items.
+	 *
+	 * @param WC_Order_Item_Product|WC_Order_Item_Fee $item     Item to process.
+	 * @param string                                  $currency Currency to use.
+	 * @return \stdClass
+	 */
+	private function process_item( WC_Order_Item $item, string $currency ): stdClass {
+		// Check to see if it is a WC_Order_Item_Product or a WC_Order_Item_Fee.
+		if ( $item instanceof WC_Order_Item_Product ) {
+			$subtotal     = $item->get_subtotal();
+			$product_id   = $item->get_variation_id()
+				? $item->get_variation_id()
+				: $item->get_product_id();
+			$product_code = substr( $product_id, 0, 12 );
+		} else {
+			$subtotal     = $item->get_total();
+			$product_code = substr( sanitize_title( $item->get_name() ), 0, 12 );
+		}
+
+		$description = substr( $item->get_name(), 0, 26 );
+		$quantity    = ceil( $item->get_quantity() );
+		$tax_amount  = $this->prepare_amount( $item->get_total_tax(), $currency );
+		if ( $subtotal >= 0 ) {
+			$unit_cost       = $this->prepare_amount( $subtotal / $quantity, $currency );
+			$discount_amount = $this->prepare_amount( $subtotal - $item->get_total(), $currency );
+		} else {
+			// It's possible to create products with negative price - represent it as free one with discount.
+			$discount_amount = abs( $this->prepare_amount( $subtotal / $quantity, $currency ) );
+			$unit_cost       = 0;
+		}
+
+		return (object) [
```

Comment:
Same idea as https://github.com/Automattic/woocommerce-payments/pull/7333/files#r1342602551

Reply:
And the same comment :)